pip_audit/_dependency_source/requirement.py

```diff
@@ -123,11 +129,30 @@ def collect(self) -> Iterator[Dependency]:
                 t.unlink()
 
     def _collect_from_files(self, filenames: list[Path]) -> Iterator[Dependency]:
+        # Figure out whether we have a fully resolved set of dependencies.
+        reqs: list[InstallRequirement] = []
+        require_hashes: bool = self._require_hashes
+        for filename in filenames:
+            rf = RequirementsFile.from_file(filename)
+            if len(rf.invalid_lines) > 0:
+                invalid = rf.invalid_lines[0]
+                raise InvalidRequirementSpecifier(
+                    f"requirement file {filename} contains invalid specifier at "
+                    f"line {invalid.line_number}: {invalid.error_message}"
+                )
+
+            # If one or more requirements have a hash, this implies `--require-hashes`.
+            require_hashes = require_hashes or any(req.hash_options for req in rf.requirements)
+            reqs.extend(rf.requirements)
+
+        # If the user has supplied `--no-deps` or there are hashed requirements, we should assume
+        # that we have a fully resolved set of dependencies and we should waste time by invoking
+        # `pip`.
+        if self._no_deps or require_hashes:
```

Comment:
@woodruffw I thought the plan was that we were going to give it some kind of scary name instead of having it activate as soon as we supply a hashed requirement since the fast path is less comprehensive than doing a real `pip` installation.

Reply:
Hi @tetsuo-cpp thanks a lot for your review.
I agree with you. In my humble opinion, invoking "preresolved dependency" based solely on the requirement presenting the hashes is a bit weak. My original idea was to add a new flag, and I explained it better in this comment on the issue. https://github.com/pypa/pip-audit/issues/610#issuecomment-1558811701

Reply:
@tetsuo-cpp You're right, I forgot about that -- IMO `--disable-pip` or similar would be appropriate.

The semantics there would be:

* `--disable-pip` will cause an error unless we're in a requirements input mode
* `--disable-pip` will cause an error unless the requirements are fully hashed *or* the user also passes `--no-deps`

Does that sound reasonable to you?

@trottomv I'd prefer to merge this as a single feature, so we'll probably want to settle the new flag before merging 🙂

Reply:
No problem for me, if I can contribute, I'm happy to do so, time permitting

Reply:
@woodruffw Yep, those flag names seem fine to me! Sounds good.

@trottomv Sorry for the churn! Are you able to make those changes? If not, I can find some time and make some pushes to this branch.

Reply:
@woodruffw Yep! Those flag names sound good to me.

@trottomv Sorry for the churn! I'm planning to push those change to your branch this afternoon and we can look at getting this released.